service/history/replication/sync_state_retriever.go

```diff
@@ -350,21 +350,18 @@ func (s *SyncStateRetrieverImpl) getNewRunInfo(ctx context.Context, namespaceId
 	}, nil
 }
 
-func (s *SyncStateRetrieverImpl) getMutation(mutableState workflow.MutableState, versionedTransition *persistencespb.VersionedTransition) (*persistencespb.WorkflowMutableStateMutation, error) {
+func (s *SyncStateRetrieverImpl) getMutation(
```

Comment:
I am separating the rewrite here into a separate PR as it fixes a race condition issue and also improves the performance of getMutation.

Reply:
https://github.com/temporalio/temporal/pull/7408